src/glossary.md

```diff
@@ -61,6 +61,11 @@ through a mechanism called ‘trait objects’.
 
 A dynamically sized type (DST) is a type without a statically known size or alignment.
 
+### Entity
+
+An [*entity*] is a [type], [item], [generic parameter], [variable binding],
```

Comment:
Hm, yea, it is difficult to define "entity" since it is a bit of a nebulous concept that is only defined by the list of things it encompasses. Some examples of how others handle it:

C++: "An entity is a value, object, reference, structured binding, function, enumerator, type, class member, bit-field,template, template specialization, namespace, or parameter pack."
[Ada](http://www.ada-auth.org/standards/12rm/html/RM-3-1.html#I1290): Doesn't directly say what an "entity" is, other than "A declaration is a language construct that associates a name with (a view of) an entity."
[Java](https://docs.oracle.com/javase/specs/jls/se11/html/jls-6.html): Defines entity as the list (similar to how I did): "A declared entity (§6.1) is a package, class type (normal or enum), …". It tends to be use the term "declared entity" more formally.
ECMA 334 (C#): Doesn't seem to ever define it, but uses it all over such as "The scope of a name is the region of program text within which it is possible to refer to the entity declared by the name without qualification of the name."
[Go](https://golang.org/ref/spec): Doesn't define it, but uses it offhand like "While the identifier of the inner declaration is in scope, it denotes the entity declared by the inner declaration."
[Haskell 2010](https://www.haskell.org/definition/haskell2010.pdf): "We use the term entity to refer to a value, type, or class defined in, imported into, or perhaps exported from a module"
[Typescript](https://github.com/microsoft/TypeScript/blob/master/doc/spec-ARCHIVED.md): Uses the term "entity" all over without explicitly defining it.

I took another stab and making it more explicit, though I think defining at as a list of things is also valid.